src/main.c

```diff
@@ -196,11 +198,31 @@ main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 
+	/* Parse wayland socket handover environment variables */
+	if (getenv("WAYLAND_SOCKET_NAME") != NULL) {
+		socket_name = strdup(getenv("WAYLAND_SOCKET_NAME"));
+		unsetenv("WAYLAND_SOCKET_NAME");
+	}
+
+	if (getenv("WAYLAND_SOCKET_FD") != NULL) {
+		socket_fd = atoi(getenv("WAYLAND_SOCKET_FD"));
+		unsetenv("WAYLAND_SOCKET_FD");
+	}
+
+	/* Fail if only one of wayland socket and fd are given */
+	if (!socket_name ^ (socket_fd == -1)) {
```

Comment:
I positively abhor the abuse of bitwise operators to effect a logical Boolean operation. I'd rather see the XOR expanded using proper logical operators, but I'd even accept

    (socket_name == NULL) != (socket_fd == -1)

because equality is a meaningful Boolean operation.

Reply:
I changed this to handle both cases being missing separately in the latest commit.